Source/Core/Core/WiiUtils.cpp

```diff
@@ -684,4 +684,89 @@ UpdateResult DoDiscUpdate(UpdateCallback update_callback, const std::string& ima
   DiscIO::NANDContentManager::Access().ClearCache();
   return result;
 }
+
+bool CheckNAND(IOS::HLE::Kernel& ios)
```

Comment:
Sorry to ask, but since it doesn't matter what part of the HAND is broken. You are going to ask permission before doing any repair anyway. Why not just return at the 'first sign' of the NAND being broken?